inference-engine/thirdparty/clDNN/src/primitive_inst.cpp

```diff
@@ -163,33 +157,38 @@ memory_impl::ptr primitive_inst::allocate_output() {
     auto use_lockable_memory = _node.is_output() || _node.get_selected_impl()->is_cpu()
                                || std::any_of(_node.get_users().begin(), _node.get_users().end(),
                                               [](const program_node* n) {return n->get_selected_impl()->is_cpu() || n->can_be_optimized(); })
-                               || engine.supports_allocation(allocation_type::usm_device) == false;
+                               || !engine.supports_allocation(allocation_type::usm_device);
     allocation_type alloc_type = use_lockable_memory ?
                                  engine.get_lockable_preffered_memory_allocation_type(layout.format.is_image_2d())
                                                      : allocation_type::usm_device;
+
+    memory::ptr res;
     if (!_network.is_internal() && (_node.can_be_optimized() || _node.is_type<generic_layer>())) {
-        return engine.allocate_memory(layout,
-                                      _node.id(),
-                                      net_id,
-                                      _node.get_memory_dependencies(),
-                                      alloc_type,
-                                      false);
+        res = engine.get_memory_from_pool(layout,
+                                           _node.id(),
+                                           net_id,
+                                           _node.get_memory_dependencies(),
+                                           alloc_type,
+                                           false);
     } else if (_network.is_internal() && _node.is_output() && _node.is_type<generic_layer>() &&
                engine.supports_allocation(allocation_type::usm_device)) {
-        return engine.allocate_memory(layout, allocation_type::usm_device, net_id, false);
+        res = engine.allocate_memory(layout, allocation_type::usm_device);
```

Comment:
Did you forget to set memory reset flag?

Reply:
fixed